lib/src/component/abstract_transition.dart

```diff
@@ -123,6 +123,9 @@ abstract class AbstractTransitionComponent<T extends AbstractTransitionProps, S
   /// Whether the Element returned by [getTransitionDomNode] will have a transition event.
   bool get hasTransition => true;
 
+  /// The duration that can elapse before a transition timeout occurs.
+  Duration get transitionTimeout => const Duration(seconds: 15);
```

Comment:
I'd say make this a prop so that it's configurable, and default it to something much smaller, like 1s, so that consumers see the warning right away.